lib/maze/runner.rb

```diff
@@ -61,18 +61,35 @@ def run_command(cmd, blocking: true, success_codes: [0])
       # @return [Array] If blocking, the output and exit_status are returned
       def run_script(script_name, blocking: false, success_codes: [0])
         script_path = File.join(SCRIPT_PATH, script_name)
-        script_path = File.join(Dir.pwd, script_name) unless File.exists? script_path
+        script_path = File.join(Dir.pwd, script_name) unless File.exist? script_path
         if Gem.win_platform?
-          # windows does not support the shebang that we use in the scripts so it
+          # Windows does not support the shebang that we use in the scripts so it
           # needs to know how to execute the script. Passing `cmd /c` tells windows
-          # to use it's known file associations to execute this path. If ruby is
-          # installed on windows then it will know that `rb` files should be exceuted
-          # using ruby etc.
+          # to use its known file associations to execute this path. If Ruby is
+          # installed on Windows then it will know that `rb` files should be executed
+          # using Ruby etc.
           script_path = "cmd /c #{script_path}"
         end
         run_command(script_path, blocking: blocking, success_codes: success_codes)
       end
 
+      # Runs a script with a given interpreter in the script directory indicated by the SCRIPT_PATH environment
+      # variable.
+      #
+      # @param script_name [String] The name of the script to run
+      # @param interpreter [String] The interpreter to use
+      # @param blocking [Boolean] Optional. Whether to wait for a return code before proceeding
+      # @param success_codes [Array] Optional. An array of integer codes which indicate the run was successful
+      #
+      # @return [Array] If blocking, the output and exit_status are returned
+      def run_interpreter(interpreter, script_name, blocking: false, success_codes: [0])
```

Comment:
I agree, but I was nervous of changing the existing implementation as it would be quite easy to break notifier tests that are reliant on it.  We have an upcoming task to review our support for running on Windows, which could well involve touching this area - so I've added a TODO to merge the new method in.  I'd rather spent a bit longer on it to make sure we don't break something in the process.

Reply:
I don't think it'd be possible to break anything with this change, but I've added a couple more comments if this method is staying

Reply:
You're right, I was being overly paranoid - and having thought about it more I don't think any notifiers that use this routine have moved up to the latest version yet anyway, reducing any risk further.